service-python/assessclaimdc6602v2/src/lib/medication.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+
+from .codesets import medication_codesets
+
+
+def categorize_med(medication_display):
+    """
+    Return the category that a medication belongs to. If it does not belong to any, return an empty list.
+
+    :param medication_display: medication text
+    :return: list
+    """
+    medication_dict = medication_codesets.med_dict
+    medication_category = []
+    for category_id in list(medication_dict.keys()):
+        if medication_category:
+            # most general category has been identified
+            break
+        for medication in medication_dict[category_id]:
+            if medication in medication_display.lower():
+                medication_category.append(category_id)
+                break
+    return medication_category
+
+
+def medication_required(request_body):
+    """
+    Determine if there is the veteran requires continuous medication for hypertension
+    :param request_body: request body
+    :type request_body: dict
+    :return: response body indicating success or failure with additional attributes
+    :rtype: dict
+    """
+    response = {}
+    relevant_medications = []
+    other_medications = []
+
+    veterans_medication = request_body["evidence"]["medications"]
+    for medication in veterans_medication:
+        if medication["status"].lower() in ["active", "on-hold", "completed", "stopped", "unknown"]:
+            medication["conditionRelated"] = "false"
+            medication_display = medication["description"]
+            category = categorize_med(medication_display)
+            medication["suggestedCategory"] = category
+            if category:
+                medication["conditionRelated"] = "true"
+                relevant_medications.append(medication)
```

Comment:
I thught only active medications mattered?

Reply:
Based on the finalized asthma report "completed" and "unknown" medications should also be included. Refer to page 20-24. 

> 

> Q: Why does the algorithm also consider the medications labeled as “completed” or “unknown”?
> A: A “completed” medication suggests that all actions that are implied by the prescription have occurred. The algorithm considers “completed” medications because adjudicators can use this information (as well as the completed medication date) to evaluate the recency and relevancy of Veterans’ asthma medications.
> An “unknown” medication is one where the source system does not know which of the status values currently applies for this observation. The algorithm includes “unknown” medications in case the medication is “active.” Adjudicators might need to review additional treatment records to make the “active” determination.
>